components/patient-data/rest/src/main/java/org/phenotips/data/rest/internal/DefaultPatientByLabeledExternalIdentifierResourceImpl.java

```diff
@@ -0,0 +1,437 @@
+            if (isLabelConfiguredByAdmin(label) || allowOtherEids()) {
+                return queryPatientsByLabelAndEid(label, id);
+            } else {
+                return Collections.emptyList();
+            }
+        } catch (Exception ex) {
+            this.logger.error("Failed to retrieve patient with label [{}] and corresponding external ID [{}]: {}",
+                label, id, ex.getMessage(), ex);
+            throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * For a given label and its corresponding id value (fields on {@code LabeledIdentifierClass} objects), query
+     * patient documents.
+     *
+     * @param label the name of the label
+     * @param id the id value for the label
+     * @return an empty list if no patients have the label identifier and its corresponding id value, else a list of
+     *         patient internal identifiers
+     * @throws Exception if there is any error during querying
+     */
+    private List<String> queryPatientsByLabelAndEid(String label, String id) throws Exception
+    {
+        Query q = null;
+        try {
+            q = this.qm.createQuery(
+                "select doc.name from Document doc, doc.object(PhenoTips.LabeledIdentifierClass) obj "
+                + "where obj.label = :label and obj.value = :value", Query.XWQL);
+            q.bindValue(KEY_LABEL, label);
+            q.bindValue(KEY_VALUE, id);
+            return q.execute();
+        } catch (QueryException ex) {
+            this.logger.warn("Failed to query patient documents with label [{}] and corresponding external ID [{}]: {}",
+                label, id, ex.getMessage(), ex);
+            throw new QueryException(ex.getMessage(), q, ex);
+        }
+    }
+
+    /**
+     * Checks if the label in question is configured at the administrative level, which means that the label is
+     * available globally for all patients. If it is, then it will be the {@code label} value of a
+     * {@code LabeledIdentifierSettings} object.
+     *
+     * @param label the name of the label
+     * @return true if the label exists, else false
+     * @throws Exception if there is any error during querying
+     */
+    private boolean isLabelConfiguredByAdmin(String label) throws Exception
+    {
+        Query q = null;
+        try {
+            q = this.qm.createQuery(
+                "select obj.label from Document doc, doc.object(PhenoTips.LabeledIdentifierSettings) obj "
+                + "where obj.label = :label", Query.XWQL);
+            q.bindValue(KEY_LABEL, label);
+            List<String> results = q.execute();
+            return !results.isEmpty();
+        } catch (QueryException ex) {
+            this.logger.warn("Failed to query LabeledIdentifierSettings object: {}", ex.getMessage(), ex);
+            throw new QueryException(ex.getMessage(), q, ex);
```

Comment:
Problems when checking if identifiers are allowed should not be propagated to the user; assume sane default values if the configuration is bad.